tests/testcase_generators/anova_oneway_testcase_generator.py

```diff
@@ -0,0 +1,130 @@
+EXPECTED_PATH = Path(
+    "tests",
+    "prod_env_tests",
+    "algorithm_validation_tests",
+    "expected",
+    "one_way_anova_expected.json",
+)
+
+
+class ANOVAOneWayTestCaseGenerator(TestCaseGenerator):
+    def compute_expected_output(self, input_data, input_parameters=None):
+        # Get data and remove missing values
+        Y, X = input_data
+        n_obs = len(Y)
+        y_names = Y.columns[0]
+        x_names = X.columns[0]
+        dict_to_df = {y_names: Y[y_names].tolist(), x_names: X[x_names].tolist()}
+        data = pd.DataFrame(dict_to_df)
+        n_groups = len(set(data[x_names]))
+
+        if n_groups < 2:
+            raise ValueError(
+                f"Not enough enums to create test case. Variable:\033[1m '{x_names}'\033[0m"
+            )
+        # Anova
+        formula = "{y} ~ {x}".format(y=y_names, x=x_names)
+        lm = ols(formula, data=data).fit()
+        aov = sm.stats.anova_lm(lm)
+        result = aov.to_dict()
+
+        data_for_min_max = data
+        min_per_group = data_for_min_max.groupby([x_names]).min()
+        max_per_group = data_for_min_max.groupby([x_names]).max()
+        var_min_per_group = min_per_group.T.to_dict(orient="records").pop()
+        var_max_per_group = max_per_group.T.to_dict(orient="records").pop()
+
+        data = pd.DataFrame()
+        data[y_names] = Y[y_names].to_numpy().squeeze()
+        data[x_names] = X[x_names].to_numpy().squeeze()
+        var_label = Y.columns.values.tolist()[0]
+        covar_label = X.columns.values.tolist()[0]
+        var_sq = y_names + "_sq"
+        data[var_sq] = Y[y_names].to_numpy() ** 2
+
+        group_stats = (
+            data[[var_label, covar_label]].groupby(covar_label).agg(["count", "sum"])
+        )
+
+        group_stats.columns = ["count", "sum"]
+        group_ssq = data[[var_sq, covar_label]].groupby(covar_label).sum()
+        group_ssq.columns = ["sum_sq"]
+        group_stats_df = pd.DataFrame(group_stats)
+        group_stats_df["group_ssq"] = group_ssq
+        cats = np.unique(data[covar_label])
+
+        categories = [c for c in cats if c in group_stats.index]
+        means = group_stats["sum"] / group_stats["count"]
+        variances = group_ssq["sum_sq"] / group_stats["count"] - means ** 2
+        sample_vars = (group_stats["count"] - 1) / group_stats["count"] * variances
+        sample_stds = np.sqrt(sample_vars)
+        df1_means_stds_dict = {
+            "categories": categories,
+            "sample_stds": list(sample_stds),
+            "means": list(means),
+        }
+        df1_means_stds = pd.DataFrame(df1_means_stds_dict, index=categories).drop(
+            "categories", 1
+        )
+        df1_means_stds["m-s"] = list(
+            df1_means_stds["means"] - df1_means_stds["sample_stds"]
+        )
+        # df1_means_stds['m'] = df1_means_stds['means']
+        df1_means_stds["m+s"] = list(
+            df1_means_stds["means"] + df1_means_stds["sample_stds"]
+        )
```

Comment:
**Important**
During algorithm validation we want to validate our implementations against some state-of-the-art implementation. The test case generators are supposed to generate test cases together with expected results based on the SOA implementation, here statsmodels and pinguin. It is therefore important to keep our code to a minimum and use the libraries for any non-trivial computation. I understand that you copied most of the generator from the corresponding one in Exareme 1. However, I don't understand where do lines 43-87 come from. It seems to me that you are repeating the computation of your own implementation, [here](https://github.com/madgik/MIP-Engine/blob/4dd85e05ba3a0b520aead68990b0a7a702b26b17/mipengine/algorithms/one_way_anova.py#L112). Why are these extra steps needed here and not in Exareme 1?

Reply:
I was trying to make sure that all results from the state-of-the-art implementation that would be used for the rest of the calculations, would produce correct results. Is there a better way to do this or should I just remove it all together?

Reply:
> I was trying to make sure that all results from the state-of-the-art implementation that would be used for the rest of the calculations, would produce correct results

We trust that the SOA implementation will produce correct results, that's why we use it.

> Is there a better way to do this or should I just remove it all together?

I'm not sure I follow. We want the test cases to be generated using **only the SOA implementation**, if possible. The corresponding test case generator in [exareme](https://github.com/madgik/exareme/blob/c0d92f705eb39989419eb96532bd524bf3a8d2e4/Exareme-Docker/src/mip-algorithms/ANOVA_ONEWAY/generate_tescases_anova.py#L40-L64) uses just the SOA implementations (statsmodels and pinguin), with one exception for the p-values because there was an inconsistency between the two SOA. 

Why are the lines 43-87 needed here but not in exareme?